tests/test_yaml/stdlib_pagmo_after_meta.yaml

```diff
@@ -0,0 +1,93 @@
+{% set version = "1.10.0" %}
+
+package:
+  name: pagmo
+  version: {{ version }}
+
+source:
+  # fake source url to get version migrator to pass
+  url: https://github.com/scipy/scipy/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 3f9e587a96844a9b4ee7f998cfe4dc3964dc95c4ca94d7de6a77bffb99f873da
+  # url: https://github.com/esa/pagmo2/archive/v{{ version }}.tar.gz
+  # sha256: 701ada528de7d454201e92a5d88903dd1c22ea64f43861d9694195ddfef82a70
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - cmake
+    - make    # [unix]
+
+outputs:
+  - name: pagmo
+    requirements:
+      host:
+        - eigen
+        - nlopt
+        - libboost-devel
+        - tbb-devel
+        - ipopt
```

Comment:
So this is a case where I'm not sure that the "no `script:`, no `build:`" rule creates the correct result, as I have often experienced that conda does _not_ propagate run-exports from the global build environment to the outputs. 

This particular case is even _more_ special because it does the "same name for feedstock and output" thing (which I consider highly fragile in practice), so it actually might work, but the question to me remains: are we 100% sure we will not create missing run-exports to `__glibc / __osx` if we _don't_ add `{{ stdlib("c") }}` in such a case?